### PR TITLE
New streamlined README.md without the spree installer

### PR DESCRIPTION
**Pull request by damianlegawiec**
Continuation of https://github.com/spree/spree/pull/6939, Spree installer won't be removed completely, but will be deprecated in the future
- [X] outdated information removed
- [X] spree installer information removed
- [X] simple gem install information added
- [X] useful links added

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,72 +137,7 @@ cd sandbox
...
-
-If you're installing this in a new Spree 1.2+ application, you'll need to install
-and run the migrations with
-
-```shell
-bundle exec rake spree_auth:install:migrations
-bundle exec rake db:migrate
-```
-
-change the following line in `config/initializers/spree.rb`
```
Comment (priyank-gupta) [inline README.md]: @damianlegawiec I think we should keep info of setting/changing `Spree.user_class` to `Spree::User` if using spree_auth_devise else dev can set whatever user class he is using.

Comment (damianlegawiec) [inline README.md]: @priyank-gupta this is only a note for legacy Spree versions (pre 2.x), using Spree with a custom setup (eg. external devise) should be covered in the guides

Comment (priyank-gupta) [inline README.md]: @damianlegawiec Makes sense.

**Comment by Mafi88** [comment]
:+1:

**Pull request merged**